arch/arm/core/isr_wrapper.S

```diff
@@ -41,6 +41,22 @@ GTEXT(_IntExit)
  */
 SECTION_FUNC(TEXT, _isr_wrapper)
 
+#ifdef CONFIG_CPU_CORTEX_R
+	/*
+	 * Save away r0-r3 from previous context to the process stack since they
+	 * are clobbered here.  Also, save away lr since we may swap processes
+	 * and return to a different thread.
+	 */
+	push {r4}
+	sub r4, lr, #4
+
+	cps #MODE_SYS
+	stmdb sp!, {r0-r4}
```

Comment:
I'm saving away r12 with stm r0, {r3-r12,sp,lr}^ in __pendsv.  I believed that was sufficient since no one touches r12 from the entry to __svc until the aforementioned instruction.

Reply:
Unfortunately, you are calling some C code before that (for example the read_timer_start_of_isr below), which is allowed to destroy R12 contents.

Similar case exist also in other places. For example `__svc` calls _irq_do_offload().

Reply:
Ok, I understand what you meant now.  I update the PR to include a fix for that.